### PR TITLE
Re-organise ESP32 boards and introduce two more.

### PR DESCRIPTION
**Pull request by Ouss4**
## Summary
This PR re-organises the ESP32 boards and introduces the ESP-WROVER-KIT and the ESP32-Ethernet-Kit.
## Impact
Other than the new organisation, this will also have a few new configs during the daily CI checks.
## Testing
All the configs build and work.

### CODE REVIEW

**Comment by Ouss4** [comment]
@btashton @xiaoxiang781216 please be aware that few more jobs will be added to the CI with this since Xtensa is included as a whole directory.

File: boards/xtensa/esp32/common/src/esp32_wdt.c [hunk 1]
```diff
@@ -23,33 +23,34 @@
  ****************************************************************************/
 
 #include <nuttx/config.h>
+
+#include <sys/types.h>
 #include <debug.h>
+
 #include "esp32_wtd_lowerhalf.h"
```
[the remark this answers is not in the document]
Comment (Ouss4) [reply boards/xtensa/esp32/common/src/esp32_wdt.c]: Yes, the renaming will come in a different PR.

File: boards/xtensa/esp32/common/include/esp32_board_tim.h [hunk 2]
```diff
@@ -18,8 +18,8 @@
  *
  ****************************************************************************/
 
-#ifndef __BOARDS_XTENSA_ESP32_COMMON_INCLUDE_TIMER_H
-#define __BOARDS_XTENSA_ESP32_COMMON_INCLUDE_TIMER_H
+#ifndef __BOARDS_XTENSA_ESP32_COMMON_INCLUDE_BOARD_TIMER_H
```
[the remark this answers is not in the document]
Comment (Ouss4) [reply boards/xtensa/esp32/common/include/esp32_board_tim.h]: Thanks. Fixed

File: boards/xtensa/esp32/common/src/esp32_wdt.c [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /****************************************************************************
- * boards/xtensa/esp32/esp32-core/src/esp32_wtd.c
+ * boards/xtensa/esp32/common/src/esp32_wtd.c
```
[the remark this answers is not in the document]
Comment (Ouss4) [reply boards/xtensa/esp32/common/src/esp32_wdt.c]: Thanks, fixed.

**Comment by Ouss4** [comment]
@xiaoxiang781216 I fixed the naming in the boards/ directory, for the rest in arch/ that will come in a different PR.

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/xtensa/esp32/esp32-devkitc/README.txt [hunk 4]
```diff
@@ -39,6 +39,9 @@ STATUS
   Espressif is working to include support to WiFi and Bluetooth, but
   it will depends on their external libraries to get it up and running.
 
+  Wifi is supported and there is a config to demonstrate its use.  Please, see
+  configs/wapi.
```
Comment (yamt) [inline boards/xtensa/esp32/esp32-devkitc/README.txt:43]: how can i prepare `/mnt/esp/wifi`?

Comment (Ouss4) [reply boards/xtensa/esp32/esp32-devkitc/README.txt:43]: @yamt There is a config for that: ESP32_WIFI_FS_MOUNPT, it has the default value of /mnt/esp/wifi.  The board bringup mounts that.

Also note that you can disable saving WIFI parameters.

**Pull request merged**